backend/src/utils/directCallUtils.ts

```diff
@@ -18,6 +19,13 @@ export const getDirectCallOptions = async (
   if (DEV_MODE) {
     // In dev mode, we always are logged in fully -- no service accounts
     headers = kubeHeaders;
+    // Fakes the call as another user to test permissions
+    if (isImpersonating()) {
+      headers = {
+        ...kubeHeaders,
+        Authorization: `Bearer ${getImpersonateAccessToken()}`,
+      };
+    }
```

Comment:
I'm not happy with the way this fails for the impersonated user when calling prometheus. So I started investigating.

Lets just ignore this use-case, and skip this step if we are calling thanos.
```suggestion
    if (isImpersonating() && !url.includes('thanos-querier-openshift-monitoring')) {
      // We are impersonating an endpoint that is not thanos -- use the token from the impersonated user
      // Thanos Querier does not grant basic user access on external routes
      headers = {
        ...kubeHeaders,
        Authorization: `Bearer ${getImpersonateAccessToken()}`,
      };
    }
```

This is related to the bug we had where basic users could not get their own status in the Dashboard, but could get it in OpenShift console. 

After a talk with the OpenShift Console folks, I learned that we were using the wrong endpoint 9091 was a special endpoint for the admin-ish users (essentially those who can list the Namespaces object). 9092 was the endpoint for talking to a namespace-locked endpoint, which basic users could use to query things like their storage value.

It appears we "need to use the thanos tenancy service, which is not exposed through a route" -- which equates to not being something we can do externally -- such as our case, in local development. There may be other ways to go about this, but effectively we can just fallback on our admin during this dev flow, as we know the call is not going to work.

For testing of permissions, we'll just need to make use of hosting it on cluster to test some prometheus things -- which is fairly easy now that all our PRs upload an image to our quay.